cmd/kubeadm/app/cmd/phases/join/kubelet.go

```diff
@@ -150,44 +165,57 @@ func runKubeletStartJoinPhase(c workflow.RunData) (returnErr error) {
 
 	// Configure the kubelet. In this short timeframe, kubeadm is trying to stop/restart the kubelet
 	// Try to stop the kubelet service so no race conditions occur when configuring it
-	klog.V(1).Infoln("[kubelet-start] Stopping the kubelet")
-	kubeletphase.TryStopKubelet()
+	if !data.DryRun() {
+		klog.V(1).Infoln("[kubelet-start] Stopping the kubelet")
+		kubeletphase.TryStopKubelet()
+	}
```

Comment:
for dryrun==true, we should mention what would happen.

```
fmt.Println("[dryrun] Would stop the kubelet")
```

Reply:
yes, I should print more info